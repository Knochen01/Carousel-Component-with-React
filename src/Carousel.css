.Carousel-title {
  align-items: center;
  justify-content: center;
  margin-left: 25%;
}
.Carousel-main {
    display: flex;
    align-items: center;
    justify-content: center;
}
.Carousel-main > i {
    cursor: pointer;
    margin: 10px;
    color: #aaa;
}
.Carousel-main > i:hover {
    color: black;
}
.Carousel-left {
  font-size: 20px;
}
.Carousel-right {
  font-size: 20px;
}
.Carousel-hidden {
  display: none;
}
