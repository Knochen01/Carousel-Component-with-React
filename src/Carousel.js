import React from 'react';
import Card from './Card';
import Carouseldata from './Database';
import { useState } from "react";
import "./Carousel.css";

const Carousel = (props) => {
    const total = Carouseldata.cartData.length;
    const [cardIdx, setCardIdx] = useState(0);
    const goFoward = () => {setCardIdx(cardIdx + 1)};
    const goBackward = () => {setCardIdx(cardIdx - 1)};
    const hideLeft = cardIdx === 0 ? "Carousel-hidden" : "";
    const hideRight = cardIdx === total - 1 ? "Carousel-hidden" : "";

    const data = Carouseldata.cartData[cardIdx];
    return (
        <>
        <h1 className="Carousel-title">{props.title}</h1>
        <div className='Carousel-main'>
        <i 
           className={`Carousel-left ${hideLeft}`}
           onClick={goBackward}
           data-testid="left-arrow">
        go back
        </i>
        <Card 
            caption={data.caption}
            src={data.src}
            currNum={cardIdx + 1}
            totalNum ={total}
        />
         <i 
            className={`Carousel-right ${hideRight}`}
            onClick={goFoward}
            data-testid="right-arrow">
        Next one
        </i>
        </div>
        </>
    )
}
export default Carousel;