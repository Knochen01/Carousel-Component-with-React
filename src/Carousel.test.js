import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import Carousel from './Carousel';

// SmokeTest
it('Should render', () => {
    render(<Carousel />)
})

// SnapshotTest
it ("should match Snapshot", () => {
    const { asFragment } = render(<Carousel />);
    expect(asFragment()).toMatchSnapshot()
})

// Test Left arrow
test ('Left arrow is working properly', () => {
    const { getByTestId, queryByAltText} = render(<Carousel />)

    const leftArrow = getByTestId("left-arrow");
    const rightArrow = getByTestId("right-arrow");
    // move to the right
    fireEvent.click(rightArrow);

    // move to the left expecting first picture to be shown
    fireEvent.click(leftArrow)
    expect(queryByAltText("Photo by Richard Pasquarella on Unsplash")).toBeInTheDocument();
    expect(queryByAltText("Photo by Richard Pasquarella on Unsplash")).toBeInTheDocument();
})

// Is left arrow missing on first
test("Left arrow is missing on first image", () => {
    const { getByTestId } = render(<Carousel />)

    const leftArrow = getByTestId("left-arrow")
    const rightArrow = getByTestId("right-arrow")
    
    expect(leftArrow).toHaveClass("Carousel-hidden")
// Move forward once
    fireEvent.click(rightArrow)
// Move forward to last 
    fireEvent.click(rightArrow)
// Is right arrow missing on last
    expect(rightArrow).toHaveClass("Carousel-hidden")
    expect(leftArrow).not.toHaveClass("Carousel-hidden")
})