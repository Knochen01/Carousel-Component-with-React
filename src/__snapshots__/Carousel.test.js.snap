// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
<DocumentFragment>
  <h1
    class="Carousel-title"
  />
  <div
    class="Carousel-main"
  >
    <i
      class="Carousel-left Carousel-hidden"
      data-testid="left-arrow"
    >
      go back
    </i>
    <div
      class="Card"
    >
      <h4
        class="Card-title"
      >
        Photo by Richard Pasquarella on Unsplash
      </h4>
      <img
        alt="Photo by Richard Pasquarella on Unsplash"
        class="Card-image"
        src="image1.jpg"
      />
      <small
        class="Card-small"
      >
        Image1 of 3
      </small>
    </div>
    <i
      class="Carousel-right "
      data-testid="right-arrow"
    >
      Next one
    </i>
  </div>
</DocumentFragment>
`;
